src/borg/testsuite/__init__.py

```diff
@@ -54,6 +55,85 @@
     st_mtime_ns_round = -4  # only >1 microsecond resolution here?
 
 
+_symlinks_supported = None
+_fifos_supported = None
+_hardlinks_supported = None
+_utime_supported = None
+
+
+def generate_tempfile_path():
+    return os.path.join(tempfile.gettempdir(), 'borg-test-' + uuid.uuid4().hex)
+
+
+def are_symlinks_supported():
+    global _symlinks_supported
+    if _symlinks_supported is None:
+        _symlinks_supported = False
+        filepath = generate_tempfile_path()
+        try:
+            os.symlink('somewhere', filepath)
+            if os.lstat(filepath) and os.readlink(filepath) == 'somewhere':
+                _symlinks_supported = True
+            os.unlink(filepath)
```

Comment:
This should be in the `finally:` branch so that it cleans the file up regardless of symlink() errors?